index.js

```diff
@@ -1,20 +1,524 @@
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
 
-start();
+/* Each room Object has:
+  roomName: name of the room,
+  roomIntro: the room's script,
+  roomActions: what verbs that can be used in the room,
+  roomConnections: what rooms you can travel to,
+  roomInventory: what objects you can interact with,
+  roomDescriptions: what combinations of roomActions and rommInventory/playerInventory is available in this room
+*/
+let outside = {
+  roomName: "Outside",
+  roomIntro:
+    "\nYou've returned home after a night on the town. You get to the door to enter the foyer but there is a keypad that needs a code and a sign right above it\n",
+  roomActions: ["enter", "help", "type", "read", "eat", "room", "i"],
+  roomConnections: ["foyer"],
+  roomInventory: ["sign", "8675309"],
+  roomDescriptions: {
+    help: function () {
+      help(outside);
+    },
+    room: function () {
+      displayRoom(outside);
+    },
+    i: function () {
+      inventory();
+    },
+    "read sign": async function () {
+      inputAnswer = await ask(
+        "\nThe code to this door is 7-digits and based off a record by Tommy Tutone in 1981\n"
+      );
+      inputFormatter(inputAnswer, currentRoom);
+    },
+    "enter foyer": async function () {
+      if (doorBool) {
+        enterRoom(foyer);
+      } else {
+        inputAnswer = await ask("\nThe door appears to be locked...\n");
+        inputFormatter(inputAnswer, currentRoom);
+      }
+    },
+    "type 8675309": async function () {
+      doorBool = true;
+      inputAnswer = await ask("\nThe door was unlocked!\n");
+      inputFormatter(inputAnswer, currentRoom);
+    },
+    "eat sandwich": function () {
+      if (playerInventory.includes("sandwich")) {
+        eatSandwich();
+      } else {
+        console.log(
+          "You had no sandwich to eat and grew irritable...\nYou Lose!"
+        );
+        process.exit();
+      }
+    },
+  },
+};
 
-async function start() {
-  const welcomeMessage = `182 Main St.
-You are standing on Main Street between Church and South Winooski.
-There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
-  let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
-  process.exit();
+let foyer = {
+  roomName: "foyer",
+  roomIntro:
+    "\nYou've stepped foot in to the foyer, empty outside of a newspaper atop a table. You see a door to the bathroom and a door to the kitchen\n",
+  roomActions: ["enter", "help", "take", "eat", "room", "i"],
+  roomConnections: ["outside", "bathroom", "kitchen"],
+  roomInventory: ["newspaper"],
+  roomDescriptions: {
+    help: function () {
+      help(foyer);
+    },
+    room: function () {
+      displayRoom(foyer);
+    },
+    i: function () {
+      inventory();
+    },
+    "take newspaper": async function () {
```

Comment:
When I took the newspaper it told me I did, however when checking my inventory it returned "nothing" and I could continue taking the newspaper from the room.